src/streaming/net/HTTPLoader.js

```diff
@@ -233,72 +253,40 @@ function HTTPLoader(cfg) {
-                config.abort(requestObject);
-            }
-        };
-
-        /**
-         * Fired when progress is terminated due to preset time expiring.
-         * @param event
-         */
-        const _ontimeout = function (event) {
-            _oncomplete();
 
-            let timeoutMessage;
-            // We know how much we already downloaded by looking at the timeout event
-            if (event.lengthComputable) {
-                let percentageComplete = (event.loaded / event.total) * 100;
-                timeoutMessage = 'Request timeout: loaded: ' + event.loaded + ', out of: ' + event.total + ' : ' + percentageComplete.toFixed(3) + '% Completed';
-            } else {
-                timeoutMessage = 'Request timeout: non-computable download size';
-            }
-            logger.warn(timeoutMessage);
-            _addHttpRequestMetric(httpRequest, httpResponse, traces);
-            _retriggerRequest();
         };
 
-        /**
-         * Fired when the request encountered an error.
-         */
-        const _onerror = function () {
-            // If we get a 404 to a media segment we should check the client clock again and perform a UTC sync in the background.
-            try {
-                if (httpResponse.status === 404 && settings.get().streaming.utcSynchronization.enableBackgroundSyncAfterSegmentDownloadError && requestObject.type === HTTPRequest.MEDIA_SEGMENT_TYPE) {
-                    // Only trigger a sync if the loading failed for the first time
-                    const initialNumberOfAttempts = mediaPlayerModel.getRetryAttemptsForType(HTTPRequest.MEDIA_SEGMENT_TYPE);
-                    if (initialNumberOfAttempts === remainingAttempts) {
-                        eventBus.trigger(Events.ATTEMPT_BACKGROUND_SYNC);
-                    }
-                }
-            } catch (e) {}
+        const _updateResourceTimingInfo = function() {
+            requestObject.startDate = requestStartTime;
+            requestObject.endDate = new Date();
+            requestObject.firstByteDate = requestObject.firstByteDate || requestStartTime;
+            httpResponse.resourceTiming.responseEnd = requestObject.endDate.getTime();
 
-            _retriggerRequest();
+            // If enabled the ResourceTimingApi we add the corresponding information to the request object.
+            // These values are more accurate and can be used by the ThroughputController later
+            _addResourceTimingValues(httpRequest, httpResponse);
         }
 
         const _loadRequest = function(loader, httpRequest, httpResponse) {
             return new Promise((resolve) => {
                 _applyRequestInterceptors(httpRequest).then((_httpRequest) => {
                     httpRequest = _httpRequest;
 
-                    httpRequest.customData.onload = _onload;
                     httpRequest.customData.onloadend = _onloadend;
-                    httpRequest.customData.onerror = _onloadend;
```

Comment:
`FetchLoader` is still using onerror in ` httpRequest.customData.onerror();`. This function will be undefined with these changes. `XHRLoader` also seems to reference `onerror`: ` xhr.onerror = httpRequest.customData.onerror;`

Reply:
In fact calling `onerror()` from `XHRLoader` or `FetchLoader` is not useful anymore since we always end up with calling `onloadend` handler in which we now control the response code to check if the request is on error. 

I completed the code cleaning in this purpose (remove `onerror` calls)

There is one use case I missed in the `FetchLoader` if there is an error when reading the stream (using `ReadableStreamDefaultReader`) but with successfull response code. I updated to handle this case.